src/ngraph_utils.h

```diff
@@ -184,8 +184,61 @@ void print_node_histogram(const std::unordered_map<string, int>&,
                           bool sorted = true);
 
 // Prints the tensor to the given output stream
-std::ostream& DumpNGTensor(std::ostream& s, const std::string& name,
-                           const std::shared_ptr<ngraph::runtime::Tensor>& t);
+template <typename T>
+std::ostream& DumpNGTensor(std::ostream& s, const string& name,
+                           const std::shared_ptr<ngraph::runtime::Tensor>& t) {
+  // std::shared_ptr<ngraph::runtime::Tensor> t{get_tensor()};
+  const ngraph::Shape& shape = t->get_shape();
+  s << "Tensor<" << name << ": ";
+  auto type = t->get_element_type();
+  bool T_is_integral = std::is_integral<T>::value;
+  bool type_is_integral = type.is_integral();
+  if (type_is_integral != T_is_integral) {
+    std::stringstream err_msg;
+    err_msg << "Tensor type " << type << " is"
+            << (type_is_integral ? " " : " not ")
+            << "integral but passed template is"
+            << (T_is_integral ? " " : " not ") << "integral";
+    throw std::invalid_argument(err_msg.str());
+  }
```

Comment:
Added some checks to prevent accidental foot-shooting in some cases, but not perfectly type safe

Reply:
It might be more useful if we do not template-ize DumpNGTensor. We can map the ng-element-type to int/float etc inside it and Call GetScalarFromTensor with that type. That way we remove the burden from the user and it is easier to use it as a utility in a loop. For eg. encapsulate has many outputs each can be of a different type. User can directly call this in loop. rather than type checking before calling

We can add it as a todo here, if this is beyond the scope of this PR. Otherwise looks good to me

Reply:
Yes, I thought of that. but could not think of any easy way of converting from ng::Type to cpp types.
One solution would be sort of a manually specified map (like `MakeConstOp` in `ngraph_builder.cc`), but for that we'd need to know the map of all ng element types to cpp types. Will add a TODO.

Atleast now, users will be made aware by the compiler that they need to supply a cpp datatype and they'd presumably take care of that instead of `DumpNgTensor` running but failing silently.

Reply:
Yes, no easy way